src/ir_passes/mono/utils.rs

```diff
@@ -0,0 +1,70 @@
+use crate::utils::Idx;
+
+/// Wraps an Idx that is meaningful in the base component
```

Comment:
Awesome!! Can we add a little more documentation for these structs to describe what exactly base and underlying components mean and how they show up in the passes?